modules/manage/pages/iceberg/iceberg-schema-changes-and-migration-guide.adoc

```diff
@@ -0,0 +1,145 @@
+= Iceberg Schema Changes and Migration Guide for Redpanda v25.3
+:description: Information about breaking schema changes for Iceberg topics in Redpanda v25.3, and actions to take when upgrading.
+:page-categories: Iceberg, Tiered Storage, Management, High Availability, Data Replication, Integration
+
+Redpanda v25.3 introduces changes that break table compatibility for Iceberg topics. Before you upgrade to v25.3, you must take specific actions to ensure that your Iceberg topics and their associated tables continue to function correctly.
+
+== Breaking changes
+
+The following table lists the schema changes introduced in Redpanda v25.3.
+
+|===
+| Field | Iceberg type translation before v25.3 | Iceberg type translation starting in v25.3 | Impact
+
+| `redpanda.timestamp` column 
+| `timestamp` type 
+| `timestamptz` (timestamp with time zone) type
+| Affects all tables created by Iceberg topics, including DLQ tables.
+
+| `redpanda.headers.key` column 
+| `binary` type 
+| `string` type 
+| Affects all tables created by Iceberg topics, including DLQ tables.
+
+| Avro optionals (two-field union of `[null, <FIELD>]`)
+| Single-field struct type
+| Optional `FIELD`
+| Affects tables created by Iceberg topics that use Avro optionals.
+
+| Avro union 
+| 
+| Field names use the type names
+| Affects tables created by Iceberg topics that use Avro unions.
+
+| Avro and Protobuf enums 
+| `integer` type
+| `string` type
+| Affects tables created by Iceberg topics that use Avro or Protobuf enums.
+
+|===
+
+== Upgrade steps
+
+When upgrading to Redpanda v25.3, you must perform these steps to migrate Iceberg topics to the new schema translation and ensure your topics continue to function correctly. Failure to perform these steps will result in data being sent to the DLQ table until you make the Iceberg tables conformant to the new schemas (step 4).
+
+. Before upgrading to v25.3, disable Iceberg on all Iceberg topics by setting the `redpanda.iceberg.mode` topic property to `disabled`. This step ensures that no additional Parquet files are written by Iceberg topics.
+. xref:upgrade:rolling-upgrade.adoc#perform-a-rolling-upgrade[Perform a rolling upgrade] to v25.3, restarting the cluster in the process.
+. Query the `GetCoordinatorState` Admin API endpoint repeatedly for these Iceberg topics to migrate to the new schema, until there are no more pending entries in the coordinator for the given topics. This step confirms that all Parquet files written pre-upgrade have been committed to the Iceberg tables.
```

Comment:
- Is this a private Admin API endpoint? This is not currently in the Admin API spec we have available to publish in docs, so we can't link out to any reference documentation for it.
- What permissions are required to query this endpoint?

Reply:
also @andrwng per my comment in the Wiki, should we flip the recommendation to polling this till it hits 0 before upgrade?   OR is it better to do it after?  Seems like if we have a choice we'd want to drain first, no?

Reply:
This is an internal endpoint given how much of the internals it exposes. I'm not sure how much detail we actually want documented for it, other than usage here.

>should we flip the recommendation to polling this till it hits 0 before upgrade

This endpoint doesn't exist pre-upgrade.